File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k % n;

        int[] arr1 = new int[k];
        int[] arr2 = new int[n-k];

        for(int i=0; i<n-k; i++){
            arr2[i]=nums[i];
        }
        for(int i=n-k; i<n; i++){
            arr1[i-(n-k)]=nums[i];
        }

        System.arraycopy(arr1, 0, nums, 0, arr1.length);
        System.arraycopy(arr2, 0, nums, arr1.length, arr2.length);





    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # n = len(nums)
        # k = k % n
        # def reverse(start: int, end: int):
        #     while start < end:
        #         nums[start], nums[end] = nums[end], nums[start]
        #         start+=1
        #         end-=1
        # reverse(0, n-1)
        # reverse(0, k-1)
        # reverse(k, n-1)  

        n = len(nums)
        k = k % n
        arr1 = []
        arr2 = []
        for i in range(n-k):
            arr1.append(nums[i])
        for i in range(n-k, n):
            arr2.append(nums[i])

        nums[:] = arr2 + arr1